src/promptflow-devkit/promptflow/_sdk/_service/apis/line_run.py

```diff
@@ -100,3 +130,30 @@ def get(self):
             line_run_ids=args.line_run_ids,
         )
         return [line_run._to_rest_object() for line_run in line_runs]
+
+
+@api.route("/search")
+class LineRunSearch(Resource):
+    @api.doc(description="Search line runs")
+    @api.marshal_list_with(line_run_model)
+    @api.response(code=200, description="Line runs")
+    def get(self):
+        client: PFClient = get_client_from_request()
+        client.traces._logger = current_app.logger
+        args = SearchLineRunParser.from_request()
+        try:
+            line_runs: typing.List[LineRunEntity] = client.traces._search_line_runs(
```

Comment:
i prefer to not add for now, otherwise, user will never query those outside the default limit?

Reply:
and currently, when user writes a query, UX will also pass that page's query (collection/session_id/run) to search API, so that this should be countable, so i prefer no default limit

Reply:
Then we should add a check that one of collection, session_id, run must exists

Reply:
make sense, I will add a limit when none of them provided

Reply:
added, may take a look again